components/style/parallel.rs

```diff
@@ -63,25 +63,34 @@ fn top_down_dom<N, C>(unsafe_nodes: UnsafeNodeList,
         // Get a real layout node.
         let node = unsafe { N::from_unsafe(&unsafe_node) };
 
+        if !context.should_process(node) {
+            continue;
+        }
+
         // Perform the appropriate traversal.
         context.process_preorder(node);
 
-        let child_count = node.children_count();
+        // Possibly enqueue the children.
+        let mut children_to_process = 0isize;
+        for kid in node.children() {
+            context.pre_process_child_hook(node, kid);
```

Comment:
Could you add a comment here explaining what `pre_process_child_hook` typically does? It'll be important for anyone reading this code to understand the flow of dirty bits.